### DISCUSSION TITLE
mSBC codec support for HFP-AG

### DISCUSSION THREAD

**Discussion by VinitMehta7492** [Arkq/bluez-alsa#694] [Q&A] [answered]
I am trying to evaluate the mSBC codec support over SCO with HFP-AG role using bluez-alsa daemon. I checked the setup instructions and found the --enable-msbc flag that needs to be enabled during build configuration. 

However, looking at the code implementation it seems the audio data encode & decode are all implemented as a part of daemon meaning if we enable the mSBC it means the bluez-alsa daemon will also use 3rd party libs to encode-decode data which seems fair. But for cases where the encoding/decoding is taken care by the controller and not host were usually the bluez-alsa daemon is running is there any configuration available to support this? I believe the ENABLE_MSBC compilation flag updates the AG negotiation flags which can be send to HF in the form of AT commands only with enabling codec negotiation. For such cases, we only need the alsa-daemon to manage the flags used for AT commands but avoid doing any encoding/decoding using 3rd party library.

Any insights on above would be really helpful.

Regards

**Comment by arkq** [comment, marked as answer]
> But for cases where the encoding/decoding is taken care by the controller and not host were usually the bluez-alsa daemon is running is there any configuration available to support this?

No, there is no such option. I have not investigated this much, but it seems that some sort of codec selection should be available (since this https://github.com/torvalds/linux/commit/904c139a2517191e48f9cb1bb2d611ae59434009) via Linux kernel: https://elixir.bootlin.com/linux/latest/source/include/net/bluetooth/bluetooth.h#L236, anyway bluez-alsa project does not implement that as for now. Any patches are welcome, though :)

> I believe the ENABLE_MSBC compilation flag updates the AG negotiation flags...

Yes, that's true.
